src/js/sorting.js

```diff
@@ -0,0 +1,71 @@
+var _ = require('underscore');
+
+var exports = module.exports = {};
+
+function copy_array(array) {
+  return array.slice();
+}
+
+function rows_to_indexers(rows) {
+  return _.range(rows.length)
+  .reverse()     // least significant first
+  .map(function(ith_row) {
+    return function(index) { return rows[ith_row][index]; };
+  });
+}
+
+exports.genomic_metric = function genomic_metric(x) {
+  var cna_order = {AMPLIFIED:400, HOMODELETED:300, GAINED:200, HEMIZYGOUSLYDELETED:100, DIPLOID: 0, undefined: 0};
```

Comment:
Might be cleaner to keep all of these single digit and then multiply it in the return value. So AMPLIFIED:4, HOMODELETED:3, ... and then return -1* (1000*cna_order[x.cna] + 100*regulated_order, ...)